commisery/versioning.py

```diff
@@ -0,0 +1,463 @@
+# Copyright (c) 2018 - 2022 TomTom N.V. (https://tomtom.com)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+from io import open
+
+import logging
+import os
+from pathlib import PurePath
+import re
+import sys
+from typing import (
+    Any,
+    IO,
+    Iterable,
+    Iterator,
+    Mapping,
+    NamedTuple,
+    NoReturn,
+    Optional,
+    Pattern,
+    Tuple,
+    Type,
+    Union,
+)
+
+if sys.version_info[:2] >= (3, 8):
+    from typing import (
+        Final,
+        Protocol,
+    )
+else:
+    from typing_extensions import (
+        Final,
+        Protocol,
+    )
+
+__all__ = (
+    'GitVersion',
+    'SemVer',
+    "Version",
+    'read_version',
+    'replace_version',
+)
+
+Stringable = Any
+TBD = Any
+log = logging.getLogger(__name__)
+
+
+class VersioningError(RuntimeError): ...
+
+
+class _IdentifierList(tuple):
+    def __str__(self) -> str:
+        return '.'.join(self)
+
+
+class Version(Protocol):
```

Comment:
just to check with @muggenhor - I proposed to use an external module for managing SemVer iso our custom implementation; what is your opinion on this?